### PR TITLE
Update B->D predictions

### PR DESCRIPTION
**Pull request by fgrosa**
- add predictions with EvtGen decayer
- modify cooking macro to automatically get original FF (to avoid mismatches with AliPhysics macro for simulation of B->D decays)
- ignore contributions with 0 weights from simulation files

### CODE REVIEW

File: models/CookFONLLPythiaPred.C [hunk 1]
```diff
@@ -31,12 +32,12 @@ enum { // options for re-weight
...
-                          std::string inFileNameCent = "DfromB_FONLLcentPythia8_FFppbar_yDcut_pp502TeV.root", // central FONLL predictions
-                          std::string inFileNameMax = "DfromB_FONLLmaxPythia8_FFppbar_yDcut_pp502TeV.root",  // max FONLL predictions
-                          std::string outFileName = "DmesonLcPredictions_502TeV_y05_FFppbar_BRPDGmix_SepContr.root",
+void CookFONLLPythiaPred(std::string inFileNameMin = "DfromB_FONLLminEvtGen_FFppbar_yDcut_pp502TeV.root", // min FONLL predictions
+                          std::string inFileNameCent = "DfromB_FONLLcentEvtGen_FFppbar_yDcut_pp502TeV.root", // central FONLL predictions
+                          std::string inFileNameMax = "DfromB_FONLLmaxEvtGen_FFppbar_yDcut_pp502TeV.root",  // max FONLL predictions
+                          std::string outFileName = "DmesonLcPredictions_502TeV_y05_EvtGen_FFee_BRPDGmix_SepContr.root",
                           int brOpt = kBRPDGmix,
-                          int ffOpt = kFFppbar,
+                          int ffOpt = kFFee,
```
Comment (fcatalan92) [inline models/CookFONLLPythiaPred.C:40]: I would set here the "default" configuration used for the D2H analysis.

**Comment by fcatalan92** [comment]
Remember to update the macro also in AliPhysics

File: models/CookFONLLPythiaPred.C (hunk 1 above)
Comment (fgrosa) [reply models/CookFONLLPythiaPred.C:40]: Ciao Fabio, yes this could be a good idea indeed (especially for the AP version). Could you do that?

Comment (fcatalan92) [reply models/CookFONLLPythiaPred.C:40]: Done

**Pull request merged**